### PR TITLE
[Teams] Fix names of Messaging Extension projects and other cleanup

### PR DESCRIPTION
**Pull request by EricDahlvang**
Remove some unused files.

Change ComposeMessagingExtensionAuth to MessagingExtensionAuth

Use GetUserTokenAsync instead of GetTokenStatusAsync (JeffD. said it is more performant)

Change ComposeMessagingExtensions to MessagingExtensionConfig.

Add manifest for search based messaging extension.

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.5/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.5",
+  "version": "1.0",
+  "id": "",
+  "packageName": "com.teams.sample.compose.extension",
```
Comment (daveta) [inline tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json]: >compose [](start = 35, length = 7)

Feel free to ignore. Not sure if we're going to keep compose as part of description - should this be just messaging?

File: tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.5/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.5",
```
Comment (daveta) [inline tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json]: >1.5 [](start = 22, length = 3)

I'm wondering if we're attempting to be mindful of what features are supported in what level?  Or just using latest?

Comment (EricDahlvang) [reply tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json]: Currently, it looks like we have manifests in the scenarios using manifestVersion 1.3, 1.5 and one with devPreview.  I've asked Andrew what we should be using.  Waiting for a response.

File: tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json (hunk 1 above)
Comment (EricDahlvang) [reply tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json]: fixed, thanks!

File: tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json (hunk 2 above)
Comment (EricDahlvang) [reply tests/Teams/MessagingExtensionAuth/TeamsAppManifest/manifest.json]: @daveta Andrew replied that manifestVersions should be 1.5, so I've included that as well.

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

**Pull request merged**